### PR TITLE
Increases code coverage of BasicWebViewClient

### PR DESCRIPTION
**Pull request by iambmelt**
This PR increases the code coverage of `BasicWebViewClient` by adding more thorough branch coverage of `shouldOverrideUrlLoading(WebView, String)` and other methods.

### CODE REVIEW

**Comment by iambmelt** [comment]
capture

Skipping testing of `logPageStartLoading(String)` because of `private` access modifier + method primarily called `Logger` instance which is tested elsewhere.

Skipping testing of `onReceivedHttpAuthRequest(WebView, HttpAuthHandler, String host, String realm)` because it primarily exercises `HttpAuthDialog` which is package `private` and cannot be mocked, spied, or otherwise injected

**Comment by iambmelt** [comment]
+ Test of SSL Error handler

capture

File: adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+import java.security.cert.X509Certificate;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.aad.adal.AuthenticationConstants.Browser.RESPONSE_ERROR_CODE;
+import static com.microsoft.aad.adal.AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE;
+
+public class BasicWebViewClientTests extends AndroidTestCase {
+
+    private static final String sNonce = "123123-123213-123";
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java]: Error Description | Line
-- | --
Name 'sNonce' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 31
Name 'sContext' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 32
Name 'sTestPkeyAuthUrl' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 33
Name 'sTestCancellationUrl' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 40
Name 'sTestExternalSiteUrl' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 42
Name 'sTestInstallRequestUrl' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. | 44

File: adal/src/androidTest/java/com/microsoft/aad/adal/JwsBuilderTests.java [hunk 2]
```diff
@@ -50,6 +50,7 @@
 
 import javax.security.auth.x500.X500Principal;
 
+import android.app.Instrumentation;
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/JwsBuilderTests.java]: This is an unused import.

File: adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java [hunk 3]
```diff
@@ -0,0 +1,454 @@
+package com.microsoft.aad.adal;
```
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java]: Need to add license information

File: adal/src/androidTest/java/com/microsoft/aad/adal/JwsBuilderTests.java (hunk 2 above)
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/JwsBuilderTests.java]: Resolved 👍

File: adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java (hunk 1 above)
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java]: Resolved 👍

File: adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java (hunk 3 above)
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/BasicWebViewClientTests.java]: Resolved

**Pull request merged**